### DISCUSSION TITLE
Inlay hints on the end of the line?

### DISCUSSION THREAD

**Discussion by darklyspaced** [mrcjkb/rustaceanvim#400] [Q&A] [answered]
Hi,

I know that in rust-tools, there was an option to display the type hints at the end of a line like so:
```rust
let y = iter
        .rev() // => Rev<Iter<'_, ...>>
        .filter(|x| *x % 2 == 0) // => Filter<Rev<Iter<'_, ...>>>
        .collect::<Vec<_>>();
```

I really don't like how the inlay hints keep moving the text as I edit my lines of code. Also, it makes complex function chains harder to follow since there is no visualisation of the types being emitted next to each line. Is there a built in option for this?

I couldn't find anything in the offical implementation of inlay-hints.

**Comment by mrcjkb** [comment, marked as answer]
- https://github.com/neovim/neovim/issues/28261

Edit: There's a plugin for it: https://github.com/chrisgrieser/nvim-lsp-endhints

**Comment by prerit714** [comment]
Hi @darklyspaced, can you share me your config file you used for rust? I would really appreciate it. Thanks.
